src/content/chainlink-functions/resources/simulation.mdx

```diff
@@ -0,0 +1,78 @@
+---
+section: chainlinkFunctions
+date: Last Modified
+title: "Simulate your Functions"
+---
+
+import { ClickToZoom, Aside } from "@components"
+
+Before making a Chainlink Functions request from your smart contract, it is always a good practice to simulate the source code off-chain to make any adjustments or corrections.
+
+Currently, there are several options for simulating a request:
+
+- [Chainlink Functions playground](https://functions.chain.link/playground).
+
+- [Chainlink Functions Hardhat Starter Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit): Use the `npx hardhat functions-simulate-script` command to simulate your Functions JavaScript source code.
+
+- [Chainlink Functions Toolkit NPM package](https://github.com/smartcontractkit/functions-toolkit): Import this NPM package into your JavaScript/TypeScript project, then use the `simulateScript` function to simulate your Functions JavaScript source code.
+
+## Chainlink Functions playground
+
+<Aside type="caution">
+  The simulation is run on the browser. There may be differences to the real Chainlink Functions environment. For
+  instance, you are limited to vanilla JavaScript and cannot use all the functionalities of Chainlink Functions, such as
```

Comment:
```suggestion
  instance, you are limited to vanilla JavaScript and cannot use all the features of Chainlink Functions, such as
```

Reply:
"functionalities of Chainlink Functions", although accurate, sounds a bit redundant. feel free to replace with a different word as you see fit. "capabilities" as you said on line 29 is also ok